creativecloud/features/firefly/firefly-interactive.js

```diff
@@ -0,0 +1,264 @@
+
+function eventOnGenerate(generateButton, media) {
+  generateButton.addEventListener('click', async () => {
+    const textToImageButton = media.querySelector('#TextToImage');
+    const textEffectsButton = media.querySelector('#TextEffects');
+    let prompt = '';
+    let promptDefault = false;
+    const userprompt = media.querySelector('.promptText')?.value;
+    const placeholderprompt = media.querySelector('.promptText')?.getAttribute('placeholder');
+    if (userprompt === '') {
+      promptDefault = true;
+      prompt = placeholderprompt;
+    } else {
+      prompt = userprompt;
+    }
+    if (textToImageButton.classList.contains('selected')) {
+      if (promptDefault) {
+        generateButton.setAttribute('daa-ll', 'SubmitTextToImage');
+      } else {
+        generateButton.setAttribute('daa-ll', 'SubmitTextToImageUserContent');
+      }
+      const { signIn } = await import('./firefly-susi.js');
+      signIn(prompt, 'goToFirefly');
+    }
+    if (textEffectsButton.classList.contains('selected')) {
+      if (promptDefault) {
+        generateButton.setAttribute('daa-ll', 'SubmitTextEffects');
+      } else {
+        generateButton.setAttribute('daa-ll', 'SubmitTextEffectsUserContent');
+      }
+      // eslint-disable-next-line import/no-cycle
+      const { signIn } = await import('./firefly-susi.js');
+      signIn(prompt, 'goToFireflyEffects');
+    }
+  });
+}
+
+function createGenFillPrompt(element) {
+  const genfillPrompt = createTag('div', { class: 'genfillPrompt' });
+  const prompt = createTag('p');
+  prompt.innerText = `${element?.innerText?.split('|')[0].split('(')[0]}`;
+  genfillPrompt.appendChild(prompt);
+  const promptText = createTag('p', { class: 'genfillPromptUsed' });
+  promptText.innerText = `${element?.innerText?.split('|')[0].split('(')[1].replaceAll(')', '')}`;
+  genfillPrompt.appendChild(promptText);
+  return genfillPrompt;
+}
+
+export default function setInteractiveFirefly(el) {
+  const buttons = el.querySelectorAll('.con-button');
+  if (buttons) {
+    buttons.forEach((button) => { if (button.innerText.includes('Firefly')) button.setAttribute('daa-ll', 'getfirefly'); });
+  }
+  const media = el.querySelector('.media');
+  const allP = media.querySelectorAll('p');
+  const enticementMode = allP[0].innerText.split('(')[1]?.replaceAll(')', '');
+  const selectorTrayMode = allP[2].innerText.split('(')[1]?.replaceAll(')', '');
+  const interactiveElemsText = Array.from(allP);
+  [...allP].forEach(async (elem) => {
+    if (elem.querySelector('strong')) {
+      elem.remove();
+      const index = interactiveElemsText.indexOf(elem);
+      interactiveElemsText.splice(index, 1);
+    }
+  });
+
+  // Remove the prompt and option text for media
+  interactiveElemsText[0].remove();
+  interactiveElemsText[1].remove();
+  interactiveElemsText[3].remove();
+  interactiveElemsText[4].remove();
+  interactiveElemsText[6].remove();
+  interactiveElemsText[7].remove();
```

Comment:
Insteap of looping on allP and then manipulating another array of all p if it requires only removing strons you can queryselect 'p strong', remove all of them and then select all p again.

Reply:
Actually I do need some values before removing the strongs like mode for Interactive selector and Enticement. hence the query slector

Reply:
As discussed we will make this change in the next PR on stage for optimization.

Reply:
This has been implemented